### ISSUE TITLE
Should automatically create a `new Model()` if you pass in a JS object in .add, .reset etc

### ISSUE THREAD

**Issue by domharrington** [bengourley/chale#5]
We dont know what type of model it will be, but it could default to a merstone model, or we could enable functionality like Backbone to provide a Model in the constructor?

**Comment by bengourley** [comment]
I am unsure about this. I didn't like where we ended up with Backbone because setting a model constructor on the collection meant that we struggled with polymorphism, specifically storing models of different type in the same collection. I know though, that this was confounded by the active record part with all the persistence/API settings.

My worry is that the collection will then be used as a model constructor, and then it will be hard to unwire that when required to do so. If we keep it simple and only allow models to be passed in, it is a little more wiring up in the common case, but keeps things open for the polymorphism case.

What do you think?

atm It's as simple as…

``` js
collection.add(data.map(function (d) { return new Model(serviceLocator, d) }))
```

**Comment by domharrington** [comment]
Backbone does handle (according to the docs) the polymorphism case http://backbonejs.org/#Collection-model.

```
var Library = Backbone.Collection.extend({

  model: function(attrs, options) {
    if (condition) {
      return new PublicDocument(attrs, options);
    } else {
      return new PrivateDocument(attrs, options);
    }
  }

});
```

But i do remember it always being not so easy to do. The only reason i raise it is because it doesnt error when you pass in an object, it just throws when you get down to https://github.com/bengourley/chale/blob/cd5c79d9ea56096f64550bbd34477cbb65de0be6/collection.js#L82 with an unhelpful error message. Maybe we can throw sooner than that if what you've passed in doesnt look like an event emitter?

**Comment by bengourley** [comment]
Yes I think throwing an error if it doesn't look like a model is the best solution. Are you happy with that?

**Comment by domharrington** [comment]
Yeah i think that'll be good.

**Comment by balaclark** [comment]
This is something that I also find useful in backbone and would like to see here. The model attribute could be optional and only cast if set?

**Comment by bengourley** [comment]
I'm going to dig my heels in again… What's wrong with the above example? Also, would you be happier if we implemented the error when a non-model is passed in?
